### PR TITLE
Hypegraph modularity2

### PR DESCRIPTION
**Pull request by pszufe**
Update do hypergraph modularity - implementation, performance, docs and testing

### CODE REVIEW

**Comment by bkamins** [comment]
I propose that I will review it once the merge conflicts are resolved. OK?

**Comment by pszufe** [comment]
> 
> 
> I propose that I will review it once the merge conflicts are resolved. OK?

of course. Ijust finished resolving.

File: src/modularity.jl [hunk 1]
```diff
@@ -1,73 +1,128 @@
 """
-    randompartition(hg::Hypergraph, n::Int)::Vector{Vector{Int}}
 
-Generates a random partition for vertices of a hypergraph `hg` into `n` subsets.
+    randompartition(h::Hypergraph, n::Int)::Vector{Set{Int}}
```
Comment (bkamins) [inline src/modularity.jl]: something is mixed up here

File: src/modularity.jl [hunk 2]
```diff
@@ -1,73 +1,128 @@
...
     end
-    deg_vs = [length(hg.v2he[i]) for i in axes(hg, 1)]
-    volV = 2*size(hg,2)
-    volP_volV = [sum(deg_vs[i] for i in p)/volV for p in partition]
-    eP = [count(i-> hes[i]>0 && (keys(hg.he2v[i]) ⊆ p), axes(hg,2)) for p in partition]
-    (sum(eP) - sum( Ed[d]*sum(volP_volV.^d) for d in 2:max_hes)) / size(hg, 2)
 end
 
+"""
+    modularity(h::Hypergraph, ha::HypergraphAggs, partition::Vector{Set{Int}})
```
Comment (bkamins) [inline src/modularity.jl]: maybe make `ha` a keyword argument and give a default calling the struct constructor?

File: src/modularity.jl (hunk 1 above)
Comment (pszufe) [reply src/modularity.jl]: looks OK in the source code.

File: src/modularity.jl [hunk 3]
```diff
@@ -1,73 +1,128 @@
...
+During the search `method.reps` random `n`-partitions will be evaluated.
+If there are many partitions having the same value the first
+one that was randomly found will be returned.
 
 Returns a `NamedTuple` where the field `bp` contains partition
 and the field `bm` contains the modularity value for that partition.
 """
-function findmodularity(hg::Hypergraph, n::Int, reps::Int)::NamedTuple{(:bp, :bm),Tuple{Vector{Vector{Int}}, Float64}}
+
+function findcommunities(h::Hypergraph, method::CFModularityRandom)::NamedTuple{(:bp, :bm),Tuple{Vector{Set{Int}}, Float64}}
```
Comment (bkamins) [inline src/modularity.jl]: why is return value annotation needed?

**Comment by bkamins** [comment]
Maybe additionally remove excess newlines in several places of the source code.

File: src/modularity.jl (hunk 1 above)
Comment (bkamins) [reply src/modularity.jl]: Sorry - I meant the one below. The docstring function signature does not math the signature in code.

**Pull request merged**